source/meta.tex

```diff
@@ -2184,44 +2186,58 @@
 forms the logical conjunction of its template type arguments.
 
 \pnum
+For the specialization \tcode{disjunction<>},
+let $\tcode{B}_{i}$ be \tcode{true_type}.
 For a specialization \tcode{conjunction<$\tcode{B}_{1}$, $\dotsc$, $\tcode{B}_{N}$>},
-if there is a template type argument $\tcode{B}_{i}$
-for which \tcode{bool($\tcode{B}_{i}$::value)} is \tcode{false},
-then instantiating \tcode{conjunction<$\tcode{B}_{1}$, $\dotsc$, $\tcode{B}_{N}$>::value}
-does not require the instantiation of \tcode{$\tcode{B}_{j}$::value} for $j > i$.
-\begin{note}
-This is analogous to the short-circuiting behavior of
-the built-in operator \tcode{\&\&}.
-\end{note}
+let $\tcode{B}_{i}$ be
+the first type in \tcode{$\tcode{B}_{1}$, $\dotsc$, $\tcode{B}_{N}$}
+for which \tcode{$\tcode{B}_{i}$::value} is \tcode{false},
+or $\tcode{B}_{N}$ if there is no such type.
 
 \pnum
-Every template type argument
-for which \tcode{$\tcode{B}_{i}$::value} is instantiated
-shall be usable as a base class and
-shall have a member \tcode{value} which
-is convertible to \tcode{bool},
-is not hidden, and
-is unambiguously available in the type.
-
-\pnum
-The specialization \tcode{conjunction<$\tcode{B}_{1}$, $\dotsc$, $\tcode{B}_{N}$>}
-has a public and unambiguous base that is either
-\begin{itemize}
-\item
-the first type $\tcode{B}_{i}$ in the list \tcode{true_type, $\tcode{B}_{1}$, $\dotsc$, $\tcode{B}_{N}$}
-for which \tcode{bool($\tcode{B}_{i}$::value)} is \tcode{false}, or
-\item
-if there is no such $\tcode{B}_{i}$, the last type in the list.
-\end{itemize}
+The specialization
+\tcode{conjunction<$\tcode{B}_{1}$, $\dotsc$, $\tcode{B}_{N}$>}
+has a public and unambiguous base class of type $\tcode{B}_{i}$.
 \begin{note}
 This means a specialization of \tcode{conjunction}
 does not necessarily inherit from
 either \tcode{true_type} or \tcode{false_type}.
 \end{note}
 
+\pnum
+Instantiation of \tcode{$\tcode{B}_{j}$::value} is required for $1 \leq j \leq i$.
+Instantiation of \tcode{$\tcode{B}_{k}$::value} is not required for $i < k \leq N$.
+\begin{note}
+This is analogous to the short-circuiting behavior of
+the built-in operator \tcode{\&\&}.
+\end{note}
+\begin{note}
+This allows types without a \tcode{value} static data member
+to be template arguments for \tcode{conjunction}, as long as an
+earlier argument evaluates to \tcode{false}, for example:
```

Comment:
Why is this not an `example` environment after the notes?

Reply:
Because the prose is clarifying the normative text, which is the purpose of a note. The statement is not specific to the example in the code, it's generally true. The example is a specific example showing it. We have lots of notes that include "for example:" like this. But I don't feel strongly about it being a note.